Tasks/JenkinsDownloadArtifacts/ArtifactDetails/JenkinsRestClient.ts

```diff
@@ -176,24 +183,52 @@ export class JenkinsRestClient {
         });
     }
 
+    private GetClient(): HttpClient {
+        tl.debug(`Creating http client`);
+
+        const endpoint = tl.getInput("serverEndpoint", true);
+        const strictSSL: boolean = ('true' !== tl.getEndpointDataParameter(endpoint, 'acceptUntrustedCerts', true));
+
+        let requestOptions: IHttpRequestOptions = {
+            ignoreSslError: strictSSL
+        };
+
+        let proxyUrl: string = tl.getVariable("agent.proxyurl");
+        if (!!proxyUrl) {
+            requestOptions['proxy'] = {
+				proxyUrl: proxyUrl, 
+				proxyUsername: tl.getVariable("agent.proxyusername"), 
+				proxyPassword: tl.getVariable("agent.proxypassword"), 
+				proxyBypassHosts: tl.getVariable("agent.proxybypasslist") ? JSON.parse(tl.getVariable("agent.proxybypasslist")) : null 
+            }
+        }
+
+        const username = tl.getEndpointAuthorizationParameter(endpoint, 'username', false);
+        const password = tl.getEndpointAuthorizationParameter(endpoint, 'password', false);
+        var handler = new handlers.BasicCredentialHandler(username, password);
+
+        return new HttpClient("JenkinsRestClient", [handler], requestOptions);
+    }
+
     public DownloadJsonContent(urlPath: string, handlebarSource: string, additionalHandlebarContext: { [key: string]: any }): Q.Promise<any> {
         let defer = Q.defer<any>();
 
         const endpoint = tl.getInput("serverEndpoint", true);
         const endpointUrl = tl.getEndpointUrl(endpoint, false);
         const jobName = tl.getInput("jobName", true);
-        const username = tl.getEndpointAuthorizationParameter(endpoint, 'username', true);
-        const password = tl.getEndpointAuthorizationParameter(endpoint, 'password', true);
         const strictSSL: boolean = ('true' !== tl.getEndpointDataParameter(endpoint, 'acceptUntrustedCerts', true));
         const jobUrlInfix = JenkinsJobDetails.GetJobUrlInfix(jobName);
 
         let requestUrl: string = `${endpointUrl}${jobUrlInfix}/${urlPath}`;
         console.log(tl.loc("DownloadingContentFromJenkinsServer", requestUrl, strictSSL));
+        let httpClient: HttpClient = this.GetClient();
+
+        httpClient.get(requestUrl).then(async (response: HttpClientResponse) => {
```

Comment:
we had some learning with mixing usage of Q.Promise & async in same function with the DownloadBuildArtifacts task.

learning was to avoid doing this & instead only do Q.Promise. can you check with @omeshp reg. this as well?